### ISSUE TITLE
DKMS compile failures on RHEL 7 with kernel 3.10.0-1062.1.2.el7.x86_64: expected identifier or ‘(’ before ‘/’ token

### ISSUE THREAD

**Issue by xtaran** [aabc/ipt-netflow#125]
Trying to install git HEAD at 664c56f on a current RHEL7 (using DKMS) fails for me as follows:

```
[ipt-netflow]# ./configure --disable-snmp-agent 
Kernel version: 3.10.0-1062.1.2.el7.x86_64 (uname)
Kernel sources: /lib/modules/3.10.0-1062.1.2.el7.x86_64/build (found)
Checking for presence of include/linux/llist.h... Yes
Checking for presence of include/linux/grsecurity.h... No
Iptables binary version: 1.4.21 (detected from /sbin/iptables)
pkg-config for version 1.4.21 exists: Yes
Check for working gcc: Yes (gcc)
Checking for presence of xtables.h... Yes (using pkg-config)
Iptables include flags:   (pkg-config)
Iptables module path: /usr/lib64/xtables (pkg-config)
Checking for DKMS... Yes.
Creating Makefile.. done.

  If you need some options enabled run ./configure --help
  Now run: make all install

[ipt-netflow]# make all install
./gen_compat_def > compat_def.h
Compiling for kernel 3.10.0-1062.1.2.el7.x86_64
make -C /lib/modules/3.10.0-1062.1.2.el7.x86_64/build M=/usr/local/src/…/ipt-netflow modules CONFIG_DEBUG_INFO=y
make[1]: Entering directory `/usr/src/kernels/3.10.0-1062.1.2.el7.x86_64'
  CC [M]  /usr/local/src/…/ipt-netflow/ipt_NETFLOW.o
In file included from /usr/local/src/…/ipt-netflow/compat.h:9:0,
                 from /usr/local/src/…/ipt-netflow/ipt_NETFLOW.c:76:
/usr/local/src/…/ipt-netflow/compat_def.h:1:1: error: expected identifier or ‘(’ before ‘/’ token
 /usr/local/src/…/ipt-netflow/cc-test-build
 ^
make[2]: *** [/usr/local/src/…/ipt-netflow/ipt_NETFLOW.o] Error 1
make[1]: *** [_module_/usr/local/src/…/ipt-netflow] Error 2
make[1]: Leaving directory `/usr/src/kernels/3.10.0-1062.1.2.el7.x86_64'
make: *** [ipt_NETFLOW.ko] Error 2
```

And `/usr/local/src/…/ipt-netflow/compat_def.h` is indeed no valid C code:

```
/usr/local/src/…/ipt-netflow/cc-test-build
// Autogenerated for /lib/modules/3.10.0-1062.1.2.el7.x86_64/build

// xt_family is declared with <linux/netfilter_ipv4/ip_tables.h>
#define HAVE_XT_FAMILY

// End of compat_def.h
```

Not sure where that first line comes from.

**Comment by aabc** [comment]
Are you sure the source is not modified? There is just nothing to output that line before `echo "// Autogenerated for $KDIR"`. And I just tested on centos 7 - no errors. Pls run `gen_compat_def` manually like `bash -x ./gen_compat_def` and show output.

**Comment by xtaran** [comment]
> Are you sure the source is not modified?

Yes, straight from the `git clone`

> There is just nothing to output that line before `echo "// Autogenerated for $KDIR"`. 

Yes and no. Found the culprit thanks to your hints where to look for it:

> And I just tested on centos 7 - no errors. Pls run `gen_compat_def` manually like `bash -x ./gen_compat_def` and show output.

It seems a local configuration issue whose source I still have to find, but the output comes from the `cd` command:

```
[…]
+ WD=cc-test-build
+ mkdir -p cc-test-build
+ cd cc-test-build
/usr/local/src/…/ipt-netflow/cc-test-build
+ echo '// Autogenerated for /lib/modules/3.10.0-1062.1.2.el7.x86_64/build'
// Autogenerated for /lib/modules/3.10.0-1062.1.2.el7.x86_64/build
[…]
```

I digged through the bash config of that host and found `CDPATH=.:..:/root`. In the login shell the behaviour persisted after I called `unset CDPATH`, but the script worked fine again afterwards.

Since this seems an official bash feature which might be used by others, too, I'm not sure if you want to just close this issue (would be ok for me) or will try to cope with environments having this setting, e.g. by unsetting it in the script beforehand or by `> /dev/null`ing all calls to `cd`.

Can also test a patch or do a pull request in case you have a proposal or preference on how to deal with this bash setting.
